cmd/gateway/zcn/gateway-zcn.go

```diff
@@ -520,6 +522,103 @@ func (zob *zcnObjects) PutObject(ctx context.Context, bucket, object string, r *
 	return
 }
 
+func (zob *zcnObjects) PutMultipleObjects(
+	ctx context.Context,
+	bucket string,
+	objects []string,
+	r []*minio.PutObjReader,
+	opts []minio.ObjectOptions,
+) ([]minio.ObjectInfo, []error) {
+	total := len(objects)
+	if total <= 0 {
+		return nil, []error{fmt.Errorf("no files to upload")}
+	}
+
+	if total != len(r) || total != len(opts) {
+		return nil, []error{fmt.Errorf("length mismatch of objects with file readers or with options")}
+	}
+
+	remotePaths := make([]string, total)
+	for i, object := range objects {
+		if bucket == rootBucketName {
+			remotePaths[i] = filepath.Join(rootPath, object)
+		} else {
+			remotePaths[i] = filepath.Join(rootPath, bucket, object)
+		}
+	}
+	operationRequests := make([]sdk.OperationRequest, total)
+	objectInfo := make([]minio.ObjectInfo, total)
+	cbs := make([]*statusCB, total)
+	for i := 0; i < total; i++ {
+		var ref *sdk.ORef
+		ref, err := getSingleRegularRef(zob.alloc, remotePaths[i])
```

Comment:
Well  if you are uploading 25 files, this will get ref from the blobbers and get consensus on it.We would see a boost in performance if done concurrently.

Reply:
I see. Will make these changes thanks.